### DISCUSSION TITLE
PCAN on mac with PCBUSB library by mac-can.com

### DISCUSSION THREAD

**Discussion by minoseigenheer** [collin80/SavvyCAN#482] [Q&A]
https://www.mac-can.com/ has a PCAN mac library which should work with the Qt Serial Bus API.
> The [Qt Serial Bus API](https://doc.qt.io/qt-5/qtserialbus-index.html) is delivered with a plug-in for PCAN interfaces by PEAK- System. Under macOS this plug-in uses the [PCBUSB library](https://www.mac-can.com/#PCBUSB_lib). You can use it to create beautiful Qt applications that run CAN and CAN FD on a Mac with PCAN-USB interfaces The [Qt CAN Bus example](https://doc.qt.io/qt-5/qtserialbus-can-example.html) runs out-of-the-box, it only requires version 0.8.1 of the PCBUSB library being installed.
> Many thanks to Miklós Márton fom Digitroll Kft. AG Electronics who patched the PCAN-Plugin for use with macOS. Great job!

I tested the [PCBUSB library](https://www.mac-can.com/#PCBUSB_lib) and it works with their [MacCAN Monitor.app](https://www.mac-can.com/#MacCAN_App)
Does anyone know how to enable QT SerialBus devices for the OSX build of SavvyCan?

**Comment by minoseigenheer** [comment]
Can anyone help me to add the QT CAN bus plugin on OSX?
Why is it not included in the OSX build?

**Reply by vincentbarat** [reply]
Hello,

I'm trying to do exactly the same.
Did you make any progress on this ?

**Reply by vincentbarat** [reply]
Don't worry. I'll check if everything works fine, and then I will write a full installation guide here.

**Reply by vincentbarat** [reply]
Well... Unfortunately, even if the PCAN device is recognized by SavvyCAN (stauts is Connected and when I configure the speed to 500000, I can see the led on the device blinking faster, meaning the PCAN receive some data (from my car) but nevertheless, nothing is displayed in the SavvyCAN interface...

I tried on Windows, the same configuration is working fine.

**Reply by vincentbarat** [reply]
YES! After recompiling SavvyCAN, everything works fine now...
I'll write a full procedure and provide some binaries for you guys.

**Comment by vincentbarat** [comment]
Folks, you can find attached an archive containing the very last version of SavvyCAN (V207 Built Sep 5 2022) and of MacCAN's PCUSB driver (0.11.2). It has been tested on macOS 12.5.1 (Monterey). An installation and a compilation procedure is provided:

```
Run precompiled SavvyCAN with Peak CAN-USB support on macOS X 12.5.1 (Monterey)
===============================================================================

1 - Install the pre-compiled MacCAN driver 0.11.2:
$ cp libPCUSB.0.11.2.dylib /usr/local/lib/libPCUSB.dylib

2 - Double click on SavvyCAN.app

Compile SavvyCAN with Peak CAN-USB support on macOS X 12.5.1 (Monterey)
=======================================================================

0 - Pre-requesite: QT 5.15.5 must be installed using HowmeBrew

1 - Download and install the MacCAN driver 0.11.2
$ wget https://github.com/mac-can/PCBUSB-Library/releases/download/v0.11.2/macOS_Library_for_PCANUSB_v0.11.2.tar.gz
$ tar xvzf macOS_Library_for_PCANUSB_v0.11.2.tar.gz
$ cd macOS_Library_for_PCANUSB_v0.11.2
$ make install

2 - Download and compile QTSerialBus 5.15.5
$ git clone https://github.com/qt/qtserialbus.git
$ cd qtserialbus
$ git checkout v5.15.5-lts-lgpl
$ qmake
$ make

3 - Download and compile the latest version of SavvyCAN
$ git clone https://github.com/collin80/SavvyCAN.git
$ cd SavvyCAN
$ qmake
$ make

4 - Build SavvyCAN bundled with all plugins and libraries includes
$ macdeployqt SavvyCAN.app
$ cp -r ../qtserialbus/plugins/canbus SavvyCAN.app/Contents/PlugIns/
$ macdeployqt SavvyCAN.app
```

[SavvyCAN V207 Built Sep 5 2022 for macOS 12.5.1 with Peak PCAN-USB device support.zip](https://github.com/collin80/SavvyCAN/files/9498085/SavvyCAN.V207.Built.Sep.5.2022.for.macOS.12.5.1.with.Peak.PCAN-USB.device.support.zip)
